### PR TITLE
add Prettier to GitHub workflows

### PR DESCRIPTION
**Pull request by kennytram**
# Description

I'm tired of seeing Prettier format issues in the dev branch while we're discussing on how we should handle our husky workflow.  Therefore, I added Prettier format automation to our GitHub workflows.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Forked the dev branch and tested it after a push. It works successfully.

### CODE REVIEW

File: .github/workflows/prettier.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Run Prettier
+        run: yarn format
+
+      - name: Commit and push changes
+        uses: stefanzweifel/git-auto-commit-action@v4
+        with:
+          commit_message: Apply Prettier changes
+          branch: ${{ github.head_ref }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (cherylli) [inline .github/workflows/prettier.yml:31]: I suggest using a more descriptive name so we know what this "GITHUB_TOKEN" is for, as we have a few other repo secrets

**Comment by cherylli** [comment]
> How Has This Been Tested?
> Not tested.

I suggest you fork this repo to your own account, to test the workflow before we merge this

File: .github/workflows/prettier.yml (hunk 1 above)
Comment (kennytram) [reply .github/workflows/prettier.yml:31]: It's automatically generated by GitHub Actions.

**Comment by kennytram** [comment]
> > How Has This Been Tested?
> > Not tested.
> 
> I suggest you fork this repo to your own account, to test the workflow before we merge this

I tested it by implementing it in a different project 2 hours ago. But alright, I'll do that.

**Comment by cherylli** [comment]
if this works we could just scrape husky? everytime I pull something from dev to my branch I (husky) had to reformat it

Or will we still try to "fix" it?

The linting part, i'm not sure if it's even that useful, I could just manually run it

**Comment by kennytram** [comment]
> if this works we could just scrape husky? everytime I pull something from dev to my branch I (husky) had to reformat it
> 
> Or will we still try to "fix" it?
> 
> The linting part, i'm not sure if it's even that useful, I could just manually run it

We can have both if we want. It's just that this Prettier format automation will "clog" the commit history every time there are format errors in the dev branch, but husky can prevent that from happening.

**Comment by Dan-Y-Ko** [comment]
> if this works we could just scrape husky? everytime I pull something from dev to my branch I (husky) had to reformat it
> 
> Or will we still try to "fix" it?
> 
> The linting part, i'm not sure if it's even that useful, I could just manually run it

Yes, because people **will** forget to run it, and then when it gets pushed and / or pr created, the build will fail lol. It's just extra time spent when it could've just been prevented with automated linting and fixing before, and people are not always going to see or notice the build failure. It's always better to have automation for this,

**Comment by cherylli** [comment]
> > if this works we could just scrape husky? everytime I pull something from dev to my branch I (husky) had to reformat it
> > Or will we still try to "fix" it?
> > The linting part, i'm not sure if it's even that useful, I could just manually run it
> 
> We can have both if we want. It's just that this Prettier format automation will "clog" the git history every time there are format errors in the dev branch, but husky can prevent that from happening.

so is it correct to say - this one also adds a commit just for the formatting? The only use of this would be make sure the dev branch is formatted if people push unformatted code?

**Comment by kennytram** [comment]
> dev branch

Yes. Because there are unexpected issues with Husky we've been having, we're never too sure that all the code pushed to the dev branch are formatted.

**Pull request closed without merging**